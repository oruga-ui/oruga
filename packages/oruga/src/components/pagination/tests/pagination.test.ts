import { describe, test, expect, afterEach } from "vitest";
import { enableAutoUnmount, mount } from "@vue/test-utils";

import OPagination from "@/components/pagination/Pagination.vue";

describe("OPagination tests", () => {
    enableAutoUnmount(afterEach);

    test("render correctly", () => {
        const wrapper = mount(OPagination);
        expect(!!wrapper.vm).toBeTruthy();
        expect(wrapper.exists()).toBeTruthy();
        expect(wrapper.attributes("data-oruga")).toBe("pagination");
        expect(wrapper.html()).toMatchSnapshot();
    });
});
