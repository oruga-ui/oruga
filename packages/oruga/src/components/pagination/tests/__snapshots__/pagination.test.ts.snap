// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`OPagination tests > render correctly 1`] = `
"<nav class="o-pag o-pag--right" data-oruga="pagination">
  <!-- 
            @slot Previous button slot
            @binding {number} number - page number 
            @binding {boolean} isCurrent - if page is current
            @binding {(event:Event): void} click - onClick handler
            @binding {string} ariaLabel - aria-label attribute
        --><button role="button" tabindex="0" class="o-pag__link o - p a g _ _ p r e v i o u s o - p a g _ _ l i n k - - d i s a b l e d" aria-label="Previous page" aria-current="false"><span class="o-icon" data-oruga="icon" aria-hidden="true"><!-- custom icon component --><!-- native css icon --><i class="mdi mdi-chevron-left mdi-24px"></i></span></button>
  <!-- 
            @slot Next button slot
            @binding {number} number - page number 
            @binding {boolean} isCurrent - if page is current
            @binding {(event:Event): void} click - onClick handler
            @binding {string} ariaLabel - aria-label attribute
        --><button role="button" tabindex="0" class="o-pag__link o - p a g _ _ n e x t o - p a g _ _ l i n k - - d i s a b l e d" aria-label="Next page" aria-current="false"><span class="o-icon" data-oruga="icon" aria-hidden="true"><!-- custom icon component --><!-- native css icon --><i class="mdi mdi-chevron-right mdi-24px"></i></span></button>
  <ul class="o-pag__list">
    <!--First-->
    <!--v-if-->
    <!--v-if-->
    <!--Pages-->
    <!--Last-->
    <!--v-if-->
    <!--v-if-->
  </ul>
</nav>"
`;
